velox/connectors/hive/tests/HivePartitionUtilTest.cpp

```diff
@@ -117,3 +125,51 @@ TEST_F(HivePartitionUtilTest, partitionName) {
         "Unsupported partition type: MAP");
   }
 }
+
+TEST_F(HivePartitionUtilTest, partitionNameForNull) {
+  RowVectorPtr input = makeRowVector(
+      {"flat_bool_col",
+       "flat_tinyint_col",
+       "flat_smallint_col",
+       "flat_int_col",
+       "flat_bigint_col",
+       "dict_string_col",
+       "const_date_col"},
+      {makeNullableFlatVector<bool>(
+           std::vector<std::optional<bool>>{std::nullopt}),
+       makeNullableFlatVector<int8_t>(
+           std::vector<std::optional<int8_t>>{std::nullopt}),
+       makeNullableFlatVector<int16_t>(
+           std::vector<std::optional<int16_t>>{std::nullopt}),
+       makeNullableFlatVector<int32_t>(
+           std::vector<std::optional<int32_t>>{std::nullopt}),
+       makeNullableFlatVector<int64_t>(
+           std::vector<std::optional<int64_t>>{std::nullopt}),
+       makeNullableDictionary<StringView>(
+           std::vector<std::optional<StringView>>{std::nullopt}),
+       makeConstant<int32_t>(std::nullopt, 1, DATE())});
+
+  std::vector<std::string> expectedPartitionKeyValues{
+      "flat_bool_col=__HIVE_DEFAULT_PARTITION__",
+      "flat_tinyint_col=__HIVE_DEFAULT_PARTITION__",
+      "flat_smallint_col=__HIVE_DEFAULT_PARTITION__",
+      "flat_int_col=__HIVE_DEFAULT_PARTITION__",
+      "flat_bigint_col=__HIVE_DEFAULT_PARTITION__",
+      "dict_string_col=__HIVE_DEFAULT_PARTITION__",
+      "const_date_col=__HIVE_DEFAULT_PARTITION__"};
+
+  std::vector<column_index_t> partitionChannels;
+  for (auto i = 1; i <= expectedPartitionKeyValues.size(); i++) {
+    partitionChannels.resize(i);
+    std::iota(partitionChannels.begin(), partitionChannels.end(), 0);
+
+    EXPECT_EQ(
+        FileUtils::makePartName(extractPartitionKeyValues(
```

Comment:
This change modifies extractPartitionKeyValues function, hence it would be nice to test just that function and its return value. FileUtils::makePartName can be tested separately to ensure that it converts empty value into __HIVE_DEFAULT_PARTITION__.